app/models/refuel.rb

```diff
@@ -1,4 +1,11 @@
 class Refuel < ApplicationRecord
   belongs_to :fuel
   belongs_to :vehicle
+
+  validates :date, presence: true
+  validates :km, presence: true, numericality: { greater_than: 0 }
+  validates :quantity, presence: true, numericality: { greater_than: 0 }
+  validates :price, presence: true, numericality: { greater_than: 0 }
```

Comment:
Float scales must be fixed (2 decimals for quantity and price, 1 decimal for km).
You can do that in the model validations, or in the db with a migration file, feel free to use your favorite method.

Reply:
Do you think it would be appropriate to change all floats in the DB into 2 steps decimals?
I mean, we also have price, km columns in other table beside this one, do we leave them be or do we change them all?

Reply:
Price has to be adapted in INVOICES, but it's not necessary to adapt km in other tables.
For REFUELS, we use the daily trip odometer which is displayed with 1 decimal on the dashborad.
But in other tables, we use the total odometer which is an integer.

Reply:
Ok, I tried to implement this by migrating but the current stored values are too big so I tried some SQL to trunk their decimal values thanks to [(https://www.techonthenet.com/postgresql/functions/trunc.php)](url) but it's not working when I add the precision, can you tell me why ?
![image](https://user-images.githubusercontent.com/58084593/107939522-25c70480-6f87-11eb-87bf-9d256dbe8e65.png)

It does what I want (kinda) without this argument
SQL : SELECT trunc(price) FROM refuels
result:
![image](https://user-images.githubusercontent.com/58084593/107939682-5dce4780-6f87-11eb-9410-2f26f717da5d.png)

Reply:
I don't know why TRUNC doesn't work with decimals, I have the same issue here.
I prefer to use ROUND, you can execute the follow ActiveRecord query in the Rails console :

Reply:
I wanted to do something different, something that could be effective in a bigger scale application/collaboration,  so that you the team only need to run the migration to update the database. I don't know if we can run activeRecord query inside a migration, I will try your suggestion as soon as I can.